plugins/auth-backend/src/service/router.ts

```diff
@@ -25,16 +28,29 @@ export interface RouterOptions {
 export async function createRouter(
   options: RouterOptions,
 ): Promise<express.Router> {
-  const logger = options.logger.child({ plugin: 'auth' });
   const router = Router();
+  const logger = options.logger.child({ plugin: 'auth' });
+
+  // configure all the providers
+  for (const providerConfig of providers) {
+    const { providerId, strategy, providerRouter } = makeProvider(
+      providerConfig,
+    );
+    logger.info(`Configuring provider: ${providerId}`);
+    passport.use(strategy);
+    router.use(`/${providerId}`, providerRouter);
+  }
+
+  passport.serializeUser((user, done) => {
+    done(null, user);
+  });
 
-  router.get('/ping', async (_req, res) => {
-    res.status(200).send('pong');
+  passport.deserializeUser((user, done) => {
+    done(null, user);
   });
 
-  const app = express();
-  app.set('logger', logger);
-  app.use(router);
+  router.use(passport.initialize());
```

Comment:
Pretty sure these need to be added before the provider middlewares? perhaps save the provider routers to an array and add them after

Reply:
Makes sense. Strange this seems to work though.